readthedocs/projects/views/private.py

```diff
@@ -1072,42 +1066,40 @@ def _search_analytics_csv_data(self):
         """Generate raw csv data of search queries."""
         project = self.get_project()
         now = timezone.now().date()
-        last_3_month = now - timezone.timedelta(days=90)
+        retention_limit = self._get_retention_days_limit(project)
+        if retention_limit in [None, -1]:
+            # Unlimited.
+            days_ago = project.pub_date.date()
+        else:
+            days_ago = now - timezone.timedelta(days=retention_limit)
 
         data = []
         if self._is_enabled(project):
             data = (
                 SearchQuery.objects.filter(
                     project=project,
-                    created__date__gte=last_3_month,
-                    created__date__lte=now,
+                    created__date__gte=days_ago,
                 )
                 .order_by('-created')
                 .values_list('created', 'query', 'total_results')
             )
 
         file_name = '{project_slug}_from_{start}_to_{end}.csv'.format(
```

Comment:
May be good to prefix the filename with `search_` so the user knows what this is data is about.

Reply:
```suggestion
        file_name = 'readthedocs_search_{project_slug_{start}_{end}.csv'.format(
```

To make it explicit.